File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/C++/_036)_Max_Sum_of_K_Consecutive_Element.cpp
#include<iostream>
using namespace std;

int maxSumOf_K_Element(int arr[], int k, int n)
{
    int max_sum=0;
    for(int i=0; i+k-1<n; i++)
    {
        int sum = 0;
        for(int j=0; j<k; j++)
            sum += arr[i+j];
        max_sum = max(sum, max_sum);
    }
    return max_sum;
}

int main()
{
    int arr[] = {1, 8, 30, -5, 20, 7}, k=3, n=6;
    cout<<"Max Sum of 3 Consecutive Element = "<<maxSumOf_K_Element(arr, k, n)<<endl;
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_036)_Max_Sum_of_K_Consecutive_Element.dart
import 'dart:math';

int maxSumOfKElements(List<int> arr, int k) {
  int maxSum = 0;

  for (int i = 0; i + k - 1 < arr.length; i++) {
    int sum = 0;
    for (int j = 0; j < k; j++) {
      sum += arr[i + j];
    }
    maxSum = max(maxSum, sum);
  }
  return maxSum;
}

void main() {
  var arr = [1, 8, 30, -5, 20, 7];
  int k = 3;
  print(
      'Maximum Sum of $k consecutive elements = ${maxSumOfKElements(arr, k)}');
}
